watchtower/blob/justice_kit.go

```diff
@@ -0,0 +1,406 @@
+package blob
+
+import (
+	"bytes"
+	"encoding/binary"
+	"errors"
+	"fmt"
+	"io"
+
+	"golang.org/x/crypto/chacha20poly1305"
+
+	"github.com/btcsuite/btcd/btcec"
+	"github.com/lightningnetwork/lnd/lnwallet"
+	"github.com/lightningnetwork/lnd/lnwire"
+)
+
+const (
+	// MinVersion is the minimum blob version supported by this package.
+	MinVersion = 0
+
+	// MaxVersion is the maximumm blob version supported by this package.
+	MaxVersion = 0
+
+	// NonceSize is the length of a chacha20poly1305 nonce, 12 bytes.
+	NonceSize = chacha20poly1305.NonceSize
+
+	// KeySize is the length of a chacha20poly1305 key, 32 bytes.
+	KeySize = chacha20poly1305.KeySize
+
+	// CiphertextExpansion is the number of bytes padded to a plaintext
+	// encrypted with chacha20poly1305, which comes from a 16-byte MAC.
+	CiphertextExpansion = 16
+
+	// V0PlaintextSize is the plaintext size of a version 0 encoded blob.
+	V0PlaintextSize = 265
+)
+
+// Size returns the size of the encoded-and-encrypted blob in bytes.
+func Size(ver uint16) int {
+	return PlaintextSize(ver) + CiphertextExpansion
+}
+
+// PlaintextSize returns the size of the encoded-but-unencrypted blob in bytes.
+func PlaintextSize(ver uint16) int {
+	switch ver {
+	case 0:
+		return V0PlaintextSize
+	default:
+		return 0
+	}
+}
+
+var (
+	// byteOrder specifies a big-endian encoding of all integer values.
+	byteOrder = binary.BigEndian
+
+	// ErrUnknownBlobVersion signals that we don't understand the requested
+	// blob encoding scheme.
+	ErrUnknownBlobVersion = errors.New("unknown blob version")
+
+	// ErrCiphertextTooSmall is a decryption error signaling that the
+	// ciphertext is smaller than the ciphertext expansion factor.
+	ErrCiphertextTooSmall = errors.New(
+		"ciphertext is too small for chacha20poly1305",
+	)
+
+	// ErrNonceSize signals that the provided nonce is improperly sized.
+	ErrNonceSize = fmt.Errorf(
+		"chacha20poly1305 nonce must be %d bytes", NonceSize,
+	)
+
+	// ErrKeySize signals that the provided key is improperly sized.
+	ErrKeySize = fmt.Errorf(
+		"chacha20poly1305 key size must be %d bytes", KeySize,
+	)
+)
+
+// PubKey is a 33-byte, serialized compressed public key.
+type PubKey [33]byte
+
+// JusticeKit is lé Blob of Justice. The JusticeKit contains information
+// required to construct a justice transaction, that sweeps a remote party's
+// revoked commitment transaction. It supports encryption and decryption using
+// chacha20poly1305, allowing the client to encrypt the contents of the blob,
+// and for a watchtower to later decrypt if action must be taken. The encoding
+// format is versioned to allow future extensions.
+type JusticeKit struct {
+	// SweepAddress is the p2wkh address of the output where the client's
```

Comment:
Why 34 bytes?

Reply:
Updated to 42 to accommodate maximally-sized witness programs